docs/en/observability/apm/configure/outputs/logstash.asciidoc

```diff
@@ -129,7 +129,14 @@ In addition to `@metadata`, APM Server provides other potentially useful fields,
 `data_stream` field, which can be used to conditionally operate on
 {observability-guide}/apm-data-model.html[event types], namespaces, or datasets.
 
-As an example, you might want to use {ls} to route all `metric` events to the same custom metrics data stream,
+// TO DO:
+// It's unclear in the issue whether it's _only_ for datasets or if it's
+// event types, namespaces, and datasets.
```

Comment:
As you aptly inferred, it applies to all the data_stream fields:

- data_stream.dataset
- data_stream.namespace
- data_stream.type

There are more fields that are `constant_keyword` and these may change over time. I think we should instruct customers to review the default templates depending on the routing they wish to use. @axw do you have any input on how to make this the least painful?

Reply:
👋  @axw can you please take a look at @marclop's question above?

Reply:
Sorry, notification got drowned out in my inbox :(

Is overriding `constant_keyword` to `keyword` something we want to encourage? It works today, but I think it goes against data stream naming. We generally assume that the data stream name matches the fields `{data_stream.type}-{data_stream.dataset}-{data_stream.namespace}`.

Could we alternatively recommend that users rename the `data_stream.dataset` field, and then rely on Logstash's [data stream routing](https://www.elastic.co/guide/en/logstash/current/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch-data-streams) functionality?

Reply:
>Could we alternatively recommend that users rename the data_stream.dataset field, and then rely on Logstash's [data stream routing](https://www.elastic.co/guide/en/logstash/current/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch-data-streams) functionality?

How would the user rename the `data_stream.dataset` field? Would the same process work for all other fields that are `constant_keyword` that @marclop mentioned above?

Reply:
@colleenmcginnis I should have said "rewrite" rather than "rename". The user would add a [`mutate` filter](https://www.elastic.co/guide/en/logstash/current/plugins-filters-mutate.html), which rewrites the `data_stream.dataset` according to whatever logic they choose. I don't have an exact Logstash configuration handy though.

Reply:
> Would the same process work for all other fields that are constant_keyword that @marclop mentioned above?

Yes, see https://www.elastic.co/guide/en/logstash/current/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch-data_stream_auto_routing:

> Automatically routes events by deriving the data stream name using specific event fields with the %{[data_stream][type]}-%{[data_stream][dataset]}-%{[data_stream][namespace]} format.

So the user could mutate any of those fields, and the output would route accordingly.